### ISSUE TITLE
Cannot access Rancher HA Host Registration URL in AWS with ELB

### ISSUE THREAD

**Issue by lekum** [rancher/rancher#4738]
**Rancher Version:** 1.0.1

**Docker Version:** 1.11.1

**OS and where are the hosts located? (cloud, bare metal, etc):**  RHEL 7.2 in AWS

**Setup Details: (single node vs. single node, internal DB vs. external DB)** Multinode, external DB (AWS Aurora), behind ELB, SSL terminated at Rancher Server

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)** Cattle

**Steps to Reproduce:** 
Followed the [instructions](http://docs.rancher.com/rancher/latest/en/installing-rancher/installing-server/multi-nodes/) for HA up to the point _Navigate to the IP or hostname of the external load balancer_

**Results:** Cannot connect to the URL, neither via HTTP or HTTPS. The :18080 interface works, however.

[rancher-ha-cattle-stdout.log.txt](https://github.com/rancher/rancher/files/255243/rancher-ha-cattle-stdout.log.txt)
[rancher-ha-stderr.log.txt](https://github.com/rancher/rancher/files/255244/rancher-ha-stderr.log.txt)
[rancher-ha-stdout.log.txt](https://github.com/rancher/rancher/files/255242/rancher-ha-stdout.log.txt)

The config of the ELB is:

```
{
    "LoadBalancerDescriptions": [
        {
            "Subnets": [
                "EDITED", 
                "EDITED", 
                "EDITED"
            ], 
            "CanonicalHostedZoneNameID": "EDITED", 
            "VPCId": "EDITED", 
            "ListenerDescriptions": [
                {
                    "Listener": {
                        "InstancePort": 18080, 
                        "LoadBalancerPort": 18080, 
                        "Protocol": "TCP", 
                        "InstanceProtocol": "TCP"
                    }, 
                    "PolicyNames": []
                }, 
                {
                    "Listener": {
                        "InstancePort": 444, 
                        "LoadBalancerPort": 443, 
                        "Protocol": "TCP", 
                        "InstanceProtocol": "TCP"
                    }, 
                    "PolicyNames": []
                }, 
                {
                    "Listener": {
                        "InstancePort": 8080, 
                        "LoadBalancerPort": 8080, 
                        "Protocol": "TCP", 
                        "InstanceProtocol": "TCP"
                    }, 
                    "PolicyNames": []
                }, 
                {
                    "Listener": {
                        "InstancePort": 81, 
                        "LoadBalancerPort": 80, 
                        "Protocol": "TCP", 
                        "InstanceProtocol": "TCP"
                    }, 
                    "PolicyNames": []
                }
            ], 
            "HealthCheck": {
                "HealthyThreshold": 2, 
                "Interval": 30, 
                "Target": "TCP:18080", 
                "Timeout": 3, 
                "UnhealthyThreshold": 2
            }, 
            "BackendServerDescriptions": [
                {
                    "InstancePort": 81, 
                    "PolicyNames": [
                        "TFEnableProxyProtocol"
                    ]
                }, 
                {
                    "InstancePort": 444, 
                    "PolicyNames": [
                        "TFEnableProxyProtocol"
                    ]
                }, 
                {
                    "InstancePort": 8080, 
                    "PolicyNames": [
                        "TFEnableProxyProtocol"
                    ]
                }, 
                {
                    "InstancePort": 18080, 
                    "PolicyNames": [
                        "TFEnableProxyProtocol"
                    ]
                }
            ], 
            "Instances": [
                {
                    "InstanceId": "EDITED"
                }, 
                {
                    "InstanceId": "EDITED"
                }, 
                {
                    "InstanceId": "EDITED"
                }
            ], 
            "DNSName": "EDITED", 
            "SecurityGroups": [
                "EDITED"
            ], 
            "Policies": {
                "LBCookieStickinessPolicies": [], 
                "AppCookieStickinessPolicies": [], 
                "OtherPolicies": [
                    "TFEnableProxyProtocol"
                ]
            }, 
            "LoadBalancerName": "aws-playground-rancher-elb", 
            "CreatedTime": "2016-05-05T14:09:50.320Z", 
            "AvailabilityZones": [
                "eu-west-1b", 
                "eu-west-1c", 
                "eu-west-1a"
            ], 
            "Scheme": "internal", 
            "SourceSecurityGroup": {
                "OwnerAlias": "894431722748", 
                "GroupName": "EDITED"
            }
        }
    ]
}

```

**Expected:** Host registration url (the one of the ELB) working via HTTPS or HTTP

**Comment by Rucknar** [comment]
@lekum Are the hosts listed as in-service in the ELB status page? Once the HA setup is complete, all your hosts should be listening on the HTTPS port, and as such should be contactable on the HTTPS URL. 

Worth noting that if your going to sit your hosts behind an ELB, you should read [this](http://docs.rancher.com/rancher/latest/en/installing-rancher/installing-server/basic-ssl-config/) doc too.

**Comment by lekum** [comment]
@Rucknar They are listening in both 81 and 444 TCP ports, however, if you try the URL, it hangs and does not get the response (also tried with a curl).

**Comment by LLParse** [comment]
@lekum There's an [open issue](https://github.com/rancher/rancher/issues/4697) related to SSL-enabled HA in AWS. I'd recommend trying HA out with plaintext (if that is an option for you).

Your ELB config looks correct _for plaintext_. If you can't access Rancher from port 80 on your LB, then I suspect a security group issue. Would you mind sharing your config? Also, could you access a UI node from :18080, navigate to the `System HA` environment, and have a look at the System Stack?

**Comment by lekum** [comment]
@LLParse The state of the management stack is:

![management_stack](https://cloud.githubusercontent.com/assets/2630720/15138999/b7e18bbc-1693-11e6-8d56-34b07566d9c6.png)

I will check the security groups again, however, from a rancher-server machine itself, if I run:

```
curl localhost:81
```

then I get:

```
curl: (52) Empty reply from server
```

Is it normal?

**Comment by vincent99** [comment]
Yes, 81 is for proxy protocol, and you are not making a request with a proxy protocol preamble.  Normal http is on 80.

**Comment by lekum** [comment]
@LLParse 

After fixing the SG, I am able to use the plaintext access thanks!

The SSL does not work, so I will keep the issue open until this aspect is solved.

Thanks again,

**Comment by LLParse** [comment]
@lekum Great! Stay tune for the SSL fix.

**Comment by deniseschannon** [comment]
Closing in favor of 4697 for the SSL fix.
